src/features/search/redux/middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { Middleware, AnyAction } from 'redux'
+import { UPDATE_TICKER } from '../redux'
+import { resetState } from 'redux/actions'
+
+export const searchMiddleware = (socket: SocketIOClient.Socket): Middleware => {
+    return ({dispatch, getState}) => {
+        return (next) => (action: AnyAction) => {
+            const { payload, type } = action
+            if (type === UPDATE_TICKER) {
+                const { favorites, charts: { range } } = getState();
+                const tickerPlusFavorites = Array.from(new Set([...favorites, payload]));
+                dispatch(resetState(undefined))
```

Comment:
Should be no need to pass `undefined` into the `resetState` action. Code smell.

Reply:
Also `resetState` sounds very imperative and not declarative. I'd look into what you expect to cause a state reset and call it that, rather than an implementation detail.

Reply:
Yes it was placed as undefined when I wasn't sure what to type a no payload action with createAction, createAction was correctly typed in a previous PR I just missed fixing this guy.

resetState is only fired for a ticker change, is tickerChange suitable? or tickerChangeStateReset better?